ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogAppendingCommitStrategy.scala

```diff
@@ -29,28 +30,27 @@ class LogAppendingCommitStrategy[Index](
       inputState: Map[DamlStateKey, Option[DamlStateValue]],
       outputState: Map[DamlStateKey, DamlStateValue],
       exporterWriteSet: Option[SubmissionAggregator.WriteSetBuilder] = None,
-  ): Future[Index] =
+  ): Future[Index] = {
+    val serializedKeyValuePairs: SortedMap[Key, Value] =
```

Comment:
Please note that we'll be doing some CPU intensive operations outside of a `Future` here (i.e., compressing values).

Reply:
True; it should become `Future { ... }` rather than `Future.successful` as it was before though.

Reply:
Yeah, I agree that this is broken but I think it's out of scope. It was broken before.